### DISCUSSION TITLE
Placing a Focussed Gaussian Beam at an Angle

### DISCUSSION THREAD

**Discussion by abigailalexjames** [epochpic/epoch#625] [Q&A]
Hi there, 

I have a question regarding setting the focussing beam at an angle.
Starting with the simple example in the quick start guide: the wavelength needs to be modified so that it becomes:

```
  lambda = laser_wavelength * cos(laser_angle) # Space wave-fronts apart correctly along k
```
Now as k also had a dependency on the wavelength the phase becomes:
` phase = -y * laser_k * sin(laser_angle) # Vary wave-front position in space`

If we now look at the Focussing beam example, we have a few new parameters to consider:
```
begin:constant
    las_k = 2.0 * pi / las_lambda    
    w0 = I_fwhm / sqrt(2.0 * loge(2.0))                  # Beam Waist
    ray_rang = pi * w0^2 / las_lambda                    # Rayleigh range
    w_boundary = w0 * sqrt(1.0 + (foc_dist/ray_rang)^2)  # Waist on boundary
    I_boundary = I_peak_Wcm2 * (w0 / w_boundary)^2       # Intens. on boundary
    rad_curve = foc_dist * (1.0 + (ray_rang/foc_dist)^2) # Boundary curv. rad.
    gouy = atan(-foc_dist/rad_curve)                     # Boundary Gouy shift
end:constant
```
```
begin:laser
    boundary = x_min
    intensity_w_cm2 = I_boundary
    lambda = las_lambda
    phase = las_k * y^2 / (2.0 * rad_curve) - gouy
    profile = gauss(y, 0, w_boundary)
end:laser
```
My query is this:
Is it still just the wavelength that needs to be corrected? The only parameters that depend on this is k and ray_range and therefore do I only need to apply the correction to these for the phase to be correct. Or, do I need to apply the dependency on all spatial variables as well, so the beam waist at boundary, the focal distance, as well as the spatial profile and the temporal profile?

**Comment by Status-Mirror** [comment]
Hey @abigailalexjames,

Focussed Gaussian beams coming in at an angle aren't something I've managed to achieve before. We give the equation of a Gaussian beam in the [Gaussian demo](https://epochpic.github.io/quickstart/basic_examples/laser.html#focussing_a_gaussian_beam). The aim of this demo is to set the electric field such that the field on the boundary matches $E(r, -x_{min})$, because if our fields match the theory at the boundary, they should propagate according to the theory and reach a focus at $x=0$

Here's the problem: if you have the beam come in at an angle, then in order to get the right fields at the boundary, you'll need to consider different boundary-to-focus distances along the entire domain. This means that terms like $w(x)$, $R_c(x)$ and $\psi(x)$ must become functions of $x$ and $y$ in EPOCH's co-ordinate system. The wavelength adjustment you mention will still need to be made, as EPOCH reads the projection of the true wave-number in the direction perpendicular to the boundary. Even if you do this, I can't guarantee the wave will propagate correctly, as EPOCH can only load $E_y$ and $E_z$, and the true wave would have an $E_x$ component.

If possible, it may be easier to extend your simulation domain, and to manually load the laser pulse into the simulation at setup using the [fields block](https://epochpic.github.io/documentation/input_deck/input_deck_fields.html). This way you can guarantee the correct fields are present, and bypass the limitations of the [laser block](https://epochpic.github.io/documentation/input_deck/input_deck_laser.html). It would probably be easiest to calculate the field values externally, then load them with binary files as outlined in [another demo](https://epochpic.github.io/quickstart/basic_examples/particle_load.html#load-particles-from-file).

Sorry I couldn't be more helpful,
Stuart
